aws/resource_aws_eks_fargate_profile.go

```diff
@@ -110,7 +110,13 @@ func resourceAwsEksFargateProfileCreate(d *schema.ResourceData, meta interface{}
 		input.Tags = keyvaluetags.New(v).IgnoreAws().EksTags()
 	}
 
-	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
+	// mutex lock for creation/deletion serialization
+	mutexKey := fmt.Sprintf("%s-fargate-profiles", d.Get("cluster_name").(string))
+	awsMutexKV.Lock(mutexKey)
+	defer awsMutexKV.Unlock(mutexKey)
+
+	// Creation of profiles must be serialized, and can take a while, increase the timeout to a long time.
+	err := resource.Retry(30*time.Minute, func() *resource.RetryError {
```

Comment:
Since the mutex handles the timing issue -- going to revert this back to the 2 minute IAM propagation timeout.